lm_eval/tasks/mlqa.py

```diff
@@ -0,0 +1,287 @@
+                           'mlqa-translate-train.zh', 'mlqa-translate-train.es', 'mlqa-translate-train.hi',
+                           'mlqa-translate-test.ar', 'mlqa-translate-test.de', 'mlqa-translate-test.vi',
+                           'mlqa-translate-test.zh',
+                           'mlqa-translate-test.es', 'mlqa-translate-test.hi']
+
+ALL_MLQA_SETS_QA = ['mlqa.ar.ar', 'mlqa.ar.de', 'mlqa.ar.vi',
+                    'mlqa.ar.zh', 'mlqa.ar.en', 'mlqa.ar.es', 'mlqa.ar.hi', 'mlqa.de.ar', 'mlqa.de.de', 'mlqa.de.vi',
+                    'mlqa.de.zh', 'mlqa.de.en', 'mlqa.de.es', 'mlqa.de.hi', 'mlqa.vi.ar', 'mlqa.vi.de', 'mlqa.vi.vi',
+                    'mlqa.vi.zh', 'mlqa.vi.en', 'mlqa.vi.es', 'mlqa.vi.hi', 'mlqa.zh.ar', 'mlqa.zh.de', 'mlqa.zh.vi',
+                    'mlqa.zh.zh', 'mlqa.zh.en', 'mlqa.zh.es', 'mlqa.zh.hi', 'mlqa.en.ar', 'mlqa.en.de', 'mlqa.en.vi',
+                    'mlqa.en.zh', 'mlqa.en.en', 'mlqa.en.es', 'mlqa.en.hi', 'mlqa.es.ar', 'mlqa.es.de', 'mlqa.es.vi',
+                    'mlqa.es.zh', 'mlqa.es.en', 'mlqa.es.es', 'mlqa.es.hi', 'mlqa.hi.ar', 'mlqa.hi.de', 'mlqa.hi.vi',
+                    'mlqa.hi.zh', 'mlqa.hi.en', 'mlqa.hi.es', 'mlqa.hi.hi']
+
+ALL_MLQA_SETS = ALL_MLQA_SETS_QA + ALL_MLQA_SETS_TRANSLATE
+
+BG_Q_A_STRINGS = {
+    "en": {"bg": "Background:", "q": "Question:", "a": "Answer:"},
+    "ar": {"bg": ":معرفتي", "q": ":سؤال", "a": ":إجابه"},
+    "de": {"bg": "Hintergrund:", "q": "Frage:", "a": "Antwort:"},
+    "zh": {"bg": "背景:", "q": "問題:", "a": "答案:"},
+    "vi": {"bg": "lý lịch:", "q": "câu hỏi:", "a": "câu trả lời:"},
+    "es": {"bg": "antecedentes:", "q": "pregunta:", "a": "respuesta:"},
+    "hi": {"bg": "Ιστορικό:", "q": "ερώτηση:", "a": "απάντηση:"},
+}
+
+
+def whitespace_tokenize(text):
+    return text.split()
+
+
+def mixed_segmentation(text):
+    segs_out = []
+    temp_str = ""
+    for char in text:
+        if re.search(r'[\u4e00-\u9fa5]', char) or char in PUNCT:
+            if temp_str != "":
+                ss = whitespace_tokenize(temp_str)
+                segs_out.extend(ss)
+                temp_str = ""
+            segs_out.append(char)
+        else:
+            temp_str += char
+
+    if temp_str != "":
+        ss = whitespace_tokenize(temp_str)
+        segs_out.extend(ss)
+
+    return segs_out
+
+
+def normalize_answer(s, lang):
+    """Lower text and remove punctuation, articles and extra whitespace."""
+
+    def remove_articles(text, lang):
+        if lang == 'en':
+            return re.sub(r'\b(a|an|the)\b', ' ', text)
+        elif lang == 'es':
+            return re.sub(r'\b(un|una|unos|unas|el|la|los|las)\b', ' ', text)
+        elif lang == 'hi':
+            return text  # Hindi does not have formal articles
+        elif lang == 'vi':
+            return re.sub(r'\b(của|là|cái|chiếc|những)\b', ' ', text)
+        elif lang == 'de':
+            return re.sub(r'\b(ein|eine|einen|einem|eines|einer|der|die|das|den|dem|des)\b', ' ', text)
```

Comment:
Case declensions for example don't get stripped which kinda messes up the normalization aspect here - for example, in German "the big dog" and "big dog" correspond to "der große Hund" and "großer Hund" respectively, so if you strip away the "der" you end up with two different strings "große Hund" and "großer Hund"

Reply:
this is copied straight from the official MLQA implementation